src/pages/Registration.tsx

```diff
@@ -0,0 +1,117 @@
+import {ArrowBackIcon, EmailIcon, Icon} from '@chakra-ui/icons'
+import {
+    Button,
+    Center,
+    FormControl,
+    FormErrorMessage,
+    FormLabel,
+    Heading, HStack,
+    Input,
+    InputGroup,
+    InputRightAddon,
+    Text,
+    useToast,
+    VStack
+} from '@chakra-ui/react'
+import {
+    Field,
+    FieldProps,
+    Form,
+    Formik,
+    FormikProps,
+    FormikValues,
+} from 'formik'
+import React from 'react'
+import { useNavigate } from 'react-router-dom'
+import { useFetch } from 'use-http'
+import * as Yup from 'yup';
+import { object, string } from 'yup'
+
+export default function Registration() {
+    const {post, response} = useFetch()
+    const navigate = useNavigate()
+    const toast = useToast()
+    const [show, setShow] = React.useState(false)
+    const handleClick = () => setShow(!show)
+
+    const register = async (values: FormikValues) => {
+        await post('/register', values)
+        if (!response.ok && response.status == 409) {
+            toast({ title: 'Email already taken - are you already registered?', description: response.status, status: response.ok ? 'success' : 'error' })
+            return
+        }
+        navigate('/dashboard')
+    }
+
+
+    const schema = object({ email: string().required().matches(/^.*((?=.*[@]){1}).*$/, "Not a valid email"),
+        password: string().required().matches(
+            /^.*(?=.{6,})((?=.*[!@#$%^&*()\-_=+{};:,<.>?]){1})(?=.*\d)((?=.*[a-z]){1})((?=.*[A-Z]){1}).*$/,
+            "Password must contain at least 6 characters, one uppercase, one number and one special case character"),
+        repeatPassword: string().required().oneOf([Yup.ref('password'), null], "Passwords don't match")
+    })
+
+    return (
+        <VStack>
+            <Center h='80vh'>
+                <Formik initialValues={{email: '', password: '', repeatPassword:''}} validationSchema={schema} onSubmit={register}>
+                    {(formProps: FormikProps<any>) =>
+                        <VStack as={Form} spacing={14} width={400}>
+                            <VStack spacing={2} width={600}>
+                                <Heading color='blue.700'>Registration</Heading>
+                                <HStack>
+                                    <Text color='blue.700' fontSize="s" align={"center"}>Register in order to gain full access to all features of GroupMatcher.</Text>
+                                </HStack>
+                            </VStack>
+                            <VStack spacing={8} width={400}>
+                                <Field name='email' children={(fieldProps: FieldProps) =>
+                                    <FormControl isInvalid={fieldProps.meta.value && fieldProps.meta.error}>
+                                        <FormLabel color='blue.700' htmlFor='email'><strong>Email address</strong></FormLabel>
```

Comment:
- The htmlFor will automatically be defined with the name='...' value passed to Field so we don't need to set it explicitely
- We can use chakra's attribute fontWeight={600} or fontWeight='bold' instead of the <strong> tag